src/hooks/use-study-path.ts

```diff
@@ -0,0 +1,132 @@
+
+                    document.title = studyName;
+                })
+                .catch((error) => {
+                    document.title = initialTitle;
+                    snackError({
+                        messageTxt: error.message,
+                        headerId: 'LoadStudyAndParentsInfoError',
+                    });
+                });
+    }, [initialTitle, snackError, studyUuid]);
+    const onStudyUpdated = useCallback(
+        (event: any) => {
+            const eventData = JSON.parse(event.data);
+            dispatch(studyUpdated(eventData));
+            if (eventData.headers) {
+                if (eventData.headers['notificationType'] === directoriesNotificationType.UPDATE_DIRECTORY) {
+                    // TODO: this receives notifications for all the public directories and all the user's private directories
+                    // At least we don't fetch everytime a notification is received, but we should instead limit the
+                    // number of notifications (they are sent to all the clients every time). Here we are only
+                    // interested in changes in parent directories of the study (study is moved, or any parent is moved
+                    // or renamed)
+                    if (studyParentDirectoriesUuidsRef.current.includes(eventData.headers['directoryUuid'])) {
+                        fetchStudyPath();
+                    }
+                }
+            }
+        },
+        [dispatch, fetchStudyPath]
+    );
+
+    useNotificationsListener(NotificationsUrlKeys.DIRECTORY, {
+        listenerCallbackMessage: onStudyUpdated,
+    });
+
+    useEffect(() => {
+        if (prevStudyPath && prevStudyPath !== studyPath) {
+            snackInfo({
+                headerId: 'moveStudyNotification',
+                headerValues: {
+                    oldStudyPath: prevStudyPath,
+                    studyPath: studyPath ?? '',
+                },
+            });
+        }
+
+        if (prevStudyName && prevStudyName !== studyName) {
+            snackInfo({
+                headerId: 'renameStudyNotification',
+                headerValues: {
+                    oldStudyName: prevStudyName,
+                    studyName: studyName ?? '',
+                },
+            });
+        }
+    }, [snackInfo, studyName, studyPath, prevStudyPath, prevStudyName]);
+
+    useEffect(() => {
+        if (!studyUuid) {
+            document.title = initialTitle;
+            return;
+        }
+        fetchStudyPath();
+    }, [studyUuid, initialTitle, fetchStudyPath]);
+
+    useEffect(() => {
+        if (studyUpdatedForce.eventData.headers) {
+            if (
+                studyUpdatedForce.eventData.headers.studyUuid === studyUuid &&
+                studyUpdatedForce.eventData.headers[UPDATE_TYPE_HEADER] === 'metadata_updated'
+            ) {
+                fetchStudyPath();
+            }
+        }
+    }, [studyUuid, studyUpdatedForce, fetchStudyPath, snackInfo]);
```

Comment:
why not using `useNotificationsListener` as well here ?

Reply:
I see it's just moved code from study-container but I think it's ok to refactor a bit in the same move as creating a custom hook

Reply:
I think you are working on refactorisation all listeners and typing event data, since this work relates only to breadcrumbs so we move only the code to this hook..

Reply:
Will be done in another PR no problem